test/integ/cli/test_cli_incremental_download.py

```
@@ -0,0 +1,501 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+Integration tests for the incremental download CLI functionality.
+"""
+
+import os
+import subprocess
+import time
+from typing import Optional, Tuple
+
+import boto3
+import pytest
+
+from deadline.client.api._job_monitoring import wait_for_job_completion
+from .job_templates import (
+    submit_dep_chain_job,
+    submit_dep_data_flow_job,
+    submit_make_many_small_files_job,
+)
+from .test_utils import DeadlineCliTest
+
+
+class IncrementalDownloadTest:
+    """
+    Class for Incremental Download Integration Tests.
+    """
+
+    def __init__(self, farm_id: str, queue_id: str):
+        self.farm_id = farm_id
+        self.queue_id = queue_id
+        self.deadline_client = boto3.client("deadline")
+
+    def wait_for_job_completion(
+        self, job_id: str, timeout: int = 600, poll_interval: int = 5
+    ) -> Tuple[bool, str]:
+        """Wait for a job to complete. Returns (isSuccess, final_status)."""
+        try:
+            result = wait_for_job_completion(
+                farm_id=self.farm_id,
+                queue_id=self.queue_id,
+                job_id=job_id,
+                timeout=timeout,
+                max_poll_interval=poll_interval,
+            )
+            return result.status == "SUCCEEDED", result.status
+        except Exception:
+            return False, "TIMEOUT"
```

Can we catch the specific timeout exception? or put the exception information in the `final_status` output string? We may lose debuggability by swallowing the original exception here.

---

Yes sure that makes sense, adding the exception to the ``final_status`` string. Getting the change out soon

---

Done